src/client/api/pool.c

```diff
@@ -29,46 +32,62 @@ daos_pool_connect(const uuid_t uuid, const char *grp,
 		return rc;
 
 	args = dc_task_get_args(task);
-	args->grp		= grp;
-	args->flags		= flags;
-	args->poh		= poh;
-	args->info		= info;
+	args->grp	= grp;
+	args->flags	= flags;
+	args->poh	= poh;
+	args->info	= info;
 	uuid_copy((unsigned char *)args->uuid, uuid);
-	args->label		= NULL;
+	args->label             = NULL;
 
 	return dc_task_schedule(task, true);
 }
 
+/**
+ * Real latest & greatest implementation of pool connect.
+ * Used by anyone including the daos_pool.h header file.
+ */
 int
-daos_pool_connect_by_label(const char *label, const char *grp,
-			   unsigned int flags, daos_handle_t *poh,
-			   daos_pool_info_t *info, daos_event_t *ev)
+daos_pool_connect2(const char *pool, const char *sys, unsigned int flags,
+		   daos_handle_t *poh, daos_pool_info_t *info, daos_event_t *ev)
 {
 	daos_pool_connect_t	*args;
 	tse_task_t		*task;
-	size_t			 label_len = 0;
+	uuid_t			 uuid;
 	int			 rc;
 
 	DAOS_API_ARG_ASSERT(*args, POOL_CONNECT);
-	if (label)
-		label_len = strnlen(label, DAOS_PROP_LABEL_MAX_LEN+1);
-	if (!label || (label_len == 0) ||
-	    (label_len > DAOS_PROP_LABEL_MAX_LEN)) {
-		D_ERROR("invalid label parameter\n");
+
+	/**
+	 * check whether it is a valid label first since this is assumed to be
+	 * the new norm
+	 */
+	if (daos_label_is_valid(pool)) {
```

Comment:
we should probably update the task API to have a single entry that is char * and move this check into the dc layer since the task API maps to the dc_* function.